Code/DistGeom/DistGeomUtils.cpp

```diff
@@ -313,64 +290,155 @@ ForceFields::ForceField *construct3DForceField(
+  \param positions       A vector of pointers to 3D Points to write out the
+  resulting coordinates
+  \param forceConstant force constant with which to constrain bond distances
+  \param numAtoms number of atoms in molecule
+
+*/
+void add12Terms(ForceFields::ForceField *ff,
+                const ForceFields::CrystalFF::CrystalFFDetails &etkdgDetails,
+                boost::dynamic_bitset<> &atomPairs,
+                RDGeom::Point3DPtrVect &positions, double forceConstant,
+                unsigned int numAtoms) {
+  for (const auto &bond : etkdgDetails.bonds) {
+    unsigned int i = bond.first;
+    unsigned int j = bond.second;
     if (i < j) {
-      atomPairs[i * N + j] = 1;
+      atomPairs[i * numAtoms + j] = 1;
     } else {
-      atomPairs[j * N + i] = 1;
+      atomPairs[j * numAtoms + i] = 1;
     }
     double d = ((*positions[i]) - (*positions[j])).length();
-    double l = d - 0.01;
-    double u = d + 0.01;
     auto *contrib = new ForceFields::UFF::DistanceConstraintContrib(
-        field, i, j, l, u, fdist);
-    field->contribs().push_back(ForceFields::ContribPtr(contrib));
+        ff, i, j, d - KNOWN_DIST_TOL, d + KNOWN_DIST_TOL, forceConstant);
+    ff->contribs().emplace_back(contrib);
   }
-
-  // 1,3 distance constraints
+}
+//! Add 1-3 distance constraints with padding at current positions to force
+/// field
+/*!
+
+  \param ff Force field to add contributions to
+  \param etkdgDetails Contains information about the ETKDG force field
+  \param atomPairs bit set for every atom pair in the molecule where
+  a bit is set to one when the atom pair is the both end atoms of a 13
+  contribution that is constrained here
+  \param positions A vector of pointers to 3D Points to write out the resulting
+  coordinates \param forceConstant force constant with which to constrain bond
+  distances \param is13Constrained bit vector with length of total num atoms of
+  the molecule where index of every central atom of improper torsion is set to
+  one \param useBasicKnowledge whether to use basic knowledge terms \param
+  numAtoms number of atoms in molecule
+
+*/
+void add13Terms(ForceFields::ForceField *ff,
+                const ForceFields::CrystalFF::CrystalFFDetails &etkdgDetails,
+                boost::dynamic_bitset<> &atomPairs,
+                RDGeom::Point3DPtrVect &positions, double forceConstant,
+                boost::dynamic_bitset<> &is13Constrained,
+                bool useBasicKnowledge, unsigned int numAtoms) {
   for (const auto &angle : etkdgDetails.angles) {
     unsigned int i = angle[0];
     unsigned int j = angle[1];
     unsigned int k = angle[2];
 
     if (i < k) {
-      atomPairs[i * N + k] = 1;
+      atomPairs[i * numAtoms + k] = 1;
     } else {
-      atomPairs[k * N + i] = 1;
+      atomPairs[k * numAtoms + i] = 1;
     }
     // check for triple bonds
-    if (angle[3]) {
+    if (useBasicKnowledge && angle[3]) {
```

Comment:
is this a change in behavior? we were not checking that `useBasicKnowledge` was set before

Reply:
No, its not a change in behavior, it's there so the `add13Terms` can be used by both  `construct3DForceField` and `constructPlain3DForceField`.